packages/@react-stately/datepicker/src/utils.ts

```diff
@@ -143,8 +138,13 @@ export function useDefaultProps(v: DateValue, granularity: Granularity): [Granul
     throw new Error('Invalid granularity ' + granularity + ' for value ' + v.toString());
   }
 
-  // If the granularity or time zone changed, update the last value.
-  if (!lastValue || lastValue[0] !== granularity || lastValue[1] !== defaultTimeZone) {
+  let [lastValue, setLastValue] = useState<[Granularity, string]>([granularity, defaultTimeZone]);
+  if (!v) {
+    return lastValue;
+  }
+
+    // If the granularity or time zone changed, update the last value.
+  if (lastValue[0] !== granularity || lastValue[1] !== defaultTimeZone) {
```

Comment:
Looks like the changes here have made it so changes to a DateField's granularity doesn't get reflected after the initial render: https://reactspectrum.blob.core.windows.net/reactspectrum/c66f9b0844320f6a8166d74881b5ec629aee0c5c/storybook/index.html?path=/story/date-and-time-datefield--default&args=granularity:hour&providerSwitcher-express=false